File: best_hsp.pl
#!/usr/bin/perl -w

use strict;
use warnings;

die "Usage:perl $0  <alignment> <evalue>  <bit_score>" if(@ARGV != 3);

my ($alignment, $max_evalue, $min_bit_score) = @ARGV;

load_alignment();

exit;

sub load_alignment{

    open(PARSER,      $alignment) || die "$!";
    
    my $init       = "";

    while(<PARSER>){
            
            chomp;
            next if(/^#/);
            my @its = split /\t/, $_;
            
            next if( $init eq $its[0] );

            my ($query, $e_value, $bit_score) = ($its[0], $its[10], $its[11]);
             
            $init  = $query;

            next if( $e_value > $max_evalue || $bit_score < $min_bit_score );
            print $_, "\n";

    }
    close PARSER;

}

File: best_hsp.py
#!/usr/bin/python

import re
import sys

blast          = sys.argv[1]
max_evalue     = float(sys.argv[2])
min_bit_score  = float(sys.argv[3])

with open(blast, 'r') as fp:
    init = ""
    for line in fp:
        if not line.startswith("#"):
            item         = line.strip().split("\t")
            evalue       = float(item[10])
            bit_score    = float(item[11])
            if init != item[0]:
                if evalue <= max_evalue and bit_score >= min_bit_score:
                    print line.strip()
                init = item[0]
